File: growing-shop-auth-before/src/main/java/com/example/growingshopauth/company/dto/CompanyResponse.java
package com.example.growingshopauth.company.dto;

import com.example.growingshopauth.company.domain.Company;
import com.example.growingshopauth.company.domain.CompanyGrade;
import lombok.*;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class CompanyResponse {

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    @AllArgsConstructor
    @Builder
    @Getter
    public static class CompanyRes {
        private Long id;
        private String name;
        private String businessRegistrationNumber;
        private CompanyGrade grade;

        public static CompanyRes from(Company company) {
            CompanyRes res = new CompanyRes();

            res.id = company.getId();
            res.name = company.getName();
            res.businessRegistrationNumber = company.getBusinessRegistrationNumber();
            res.grade = company.getGrade();

            return res;
        }
    }
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/company/dto/CompanyResponse.kt
package com.example.growingshopauth.company.dto

import com.example.domain.company.Company
import com.example.domain.company.CompanyGrade

class CompanyResponse {

    class CompanyRes(
        val id: Long,
        val name: String,
        val businessRegistrationNumber: String,
        val grade: CompanyGrade
    ) {
        companion object {
            fun from(company: Company): CompanyRes {
                return CompanyRes(
                    company.id!!,
                    company.name,
                    company.businessRegistrationNumber,
                    company.grade
                )
            }
        }
    }
}
